server/src/main/java/org/elasticsearch/cluster/coordination/Join.java

```diff
@@ -16,59 +16,65 @@
 
 /**
  * Triggered by a {@link StartJoinRequest}, instances of this class represent join votes,
- * and have a source and target node. The source node is the node that provides the vote,
- * and the target node is the node for which this vote is cast. A node will only cast
- * a single vote per term, and this for a unique target node. The vote also carries
- * information about the current state of the node that provided the vote, so that
- * the receiver of the vote can determine if it has a more up-to-date state than the
- * source node.
+ * and have a voting and candidate-master node. The voting node is the node that provides
+ * the vote, and the candidate-master node is the node for which this vote is cast. A join
```

Comment:
nit: suggest aligning wording with field name

```suggestion
 * and have a voting and master-candidate node. The voting node is the node that provides
 * the vote, and the master-candidate node is the node for which this vote is cast. A join
```